src/types/schema.d.ts

```diff
@@ -0,0 +1,100 @@
+import type { UUID } from 'crypto';
+
+export interface Profile {
+  user_id: UUID;
+  first_name: string;
+  last_name: string;
+  eoir_registered?: boolean;
+  hours_per_month: number;
+  bar_number?: string;
+  start_date: string; // timestamp
+  availability_description?: string;
+  expected_bar_date?: string; // timestamp
+  location: string; // city, ST
+  phone_number: string;
+  state_barred?: string;
+}
+
+export type RoleEnum = 'ATTORNEY' | 'INTERPRETER' | 'LEGAL_FELLOW';
+
+export interface ProfileLanguage {
+  user_id: UUID;
+  language_name: string;
+  can_read: boolean;
+  can_speak: boolean;
+}
+
+export interface ProfileRole {
+  user_id: UUID;
+  role: RoleEnum;
+}
+
+export type ListingType = 'CASE' | 'CASE_INT' | 'LCA' | 'DOC' | 'INT';
+
+export interface Interest {
+  listing_id: UUID;
+  user_id: UUID;
+  listing_type: ListingType;
+  form_response: {
+    needs_interpreter?: boolean; // for attorneys only
+    interest_reason?: string; // for everything but non-case language support
+    start_date?: Date; // for cases and case interpretation
+  };
+}
+
+export interface CaseListing {
+  id: UUID;
+  legal_server_id: string;
+  title?: string;
+  needs_attorney?: boolean;
+  needs_interpreter?: boolean;
+  summary?: string;
+  country?: string;
+  client_location?: string;
+  hours_per_week?: number;
+  num_weeks?: number;
+  is_remote?: boolean;
+  adjudicating_agency?: string;
+  upcoming_date?: string; // timestamp
+  is_detained?: boolean;
+  languages: string[];
+  relief_codes: string[];
+  listing_type: 'CASE';
+}
+
+export interface LimitedCaseAssignment {
+  id: UUID;
+  title: string;
+  summary: string;
+  country: string;
+  deadline: string; // timestamp (deadline to submit assignment)
+  research_topic: string;
+  deliverable: string; // expected deliverable
+  languages: string[];
+  listing_type: 'LCA';
+}
+
+export interface Interpretation {
+  id: UUID;
+  title: string;
+  summary: string;
+  is_remote: boolean;
+  languages: string[];
+  listing_type: 'INT';
+}
+
+export interface DocumentTranslation {
+  id: UUID;
+  title: string;
+  summary: string;
+  deadline: string;
+  languages: string[];
+  num_pages: number;
+  listing_type: 'DOC';
+}
+
+export type Listing =
+  | CaseListing
+  | Interpretation
+  | LimitedCaseAssignment
+  | Document;
```

Comment:
we probably wanna update this to `DocumentTranslation`

Reply:
oh yea.